relay-general/src/store/transactions/processor.rs

```diff
@@ -17,15 +20,30 @@ pub struct TransactionNameConfig<'r> {
     pub rules: &'r [TransactionNameRule],
 }
 
+#[derive(Default)]
+pub struct MeasurementNameConfig {
+    /// How long the name-part of the MRI can be.
+    pub max_len: Option<usize>,
```

Comment:
No need to keep track of two numbers, a single budget for name + unit should be enough:

```suggestion
    /// Combined maximum length of the measurement name + unit.
    pub max_name_unit_len: Option<usize>,
```